cpp/dolfinx/fem/interpolate.h

```diff
@@ -598,8 +629,252 @@ void interpolate(Function<T>& u, const Function<T>& v,
     // Get mesh and check that functions share the same mesh
     if (mesh != v.function_space()->mesh())
     {
-      throw std::runtime_error(
-          "Interpolation on different meshes not supported (yet).");
+      const int nProcs = dolfinx::MPI::size(MPI_COMM_WORLD);
+      const auto mpi_rank = dolfinx::MPI::rank(MPI_COMM_WORLD);
+
+      const int tdim = u.function_space()->mesh()->topology().dim();
+      const auto cell_map
+          = u.function_space()->mesh()->topology().index_map(tdim);
+      const int num_cells = cell_map->size_local() + cell_map->num_ghosts();
+      std::vector<std::int32_t> cells(num_cells, 0);
+      std::iota(cells.begin(), cells.end(), 0);
+
+      // Collect all the points at which values are needed to define the
+      // interpolating function
+      const xt::xtensor<double, 2> x = fem::interpolation_coords(
+          *u.function_space()->element(), *u.function_space()->mesh(), cells);
+
+      // This transposition is a quick and dirty solution and should be avoided
+      auto x_t = xt::zeros_like(xt::transpose(x));
+      for (decltype(x.shape(1)) i = 0; i < x.shape(1); ++i)
+      {
+        for (decltype(x.shape(0)) j = 0; j < x.shape(0); ++j)
+        {
+          x_t(i, j) = x(j, i);
+        }
+      }
+
+      // Create a global bounding-box tree to quickly look for processes that
+      // might be able to evaluate the interpolating function at a given point
+      dolfinx::geometry::BoundingBoxTree bb(*v.function_space()->mesh(), tdim,
+                                            0.0001);
+      auto globalBB = bb.create_global_tree(MPI_COMM_WORLD);
+
+      // For each point that needs to be evaluated, store a list of processes
+      // that might be able to do it
+      std::vector<std::vector<std::int32_t>> candidates(x_t.shape(0));
+      for (decltype(x_t.shape(0)) i = 0; i < x_t.shape(0); ++i)
+      {
+        const auto collisions = dolfinx::geometry::compute_collisions(
+            globalBB,
+            xt::xtensor<double, 2>({{x_t(i, 0), x_t(i, 1), x_t(i, 2)}}));
+        candidates[i] = std::vector<std::int32_t>(collisions.links(0).cbegin(),
+                                                  collisions.links(0).cend());
+      }
```

Comment:
As `compute_collisions` (https://github.com/FEniCS/dolfinx/blob/c30adac64e5f5a19e7909666d18a8d6d8d38ebfb/cpp/dolfinx/geometry/utils.h#L50-L53) gives you an adjacency list, this is not necessary.
I would call:
```
const auto collisions = dolfinx::geometry::compute_collisions(globalBB, x_t);
```
where `collisions.links(i)` gives you the processes which collide with the ith point.

Reply:
Done. I was temporarily overkilling it with the copying to be on the safe side, thanks!